syntax = "proto3";

package messages;

// Handshake messages

message ClientHello{
    bytes nonce = 1;
    bytes key_share = 2;
    bytes share_signature = 3;
}

message ServerHello{
    bytes nonce = 1;
    bytes key_share = 2;
}

message VerifyIntegrity{
    bytes integrity = 1;
}

message EncryptedData{
    bytes nonce = 1;
    bytes encrypted = 2;
}

message Status{
    enum State{
        OK = 0;
        ERROR = 1;
    }
    string message = 3;
    State state = 4;
}

// rpc Endpoint

service Server{
    rpc InitiateHandshake(ClientHello) returns (ServerHello){}
    rpc Integrity(VerifyIntegrity) returns (VerifyIntegrity){}
    rpc TransferData(EncryptedData) returns (Status){} 
}